controller/engine_controller.go

```diff
@@ -1415,11 +1458,23 @@ func (ec *EngineController) Upgrade(e *longhorn.Engine) (err error) {
 	}()
 
 	log := ec.logger.WithField("volume", e.Spec.VolumeName)
-	client, err := GetClientForEngine(e, ec.engines, e.Spec.EngineImage)
+
+	engineCliClient, err := GetClientForEngine(e, ec.engines, e.Spec.EngineImage)
+	if err != nil {
+		return err
+	}
+
+	im, err := ec.ds.GetInstanceManager(e.Status.InstanceManagerName)
 	if err != nil {
 		return err
 	}
-	version, err := client.Version(false)
+
+	engineClientProxy, err := ec.proxyHandler.GetClient(im, engineCliClient)
+	if err != nil {
+		return err
+	}
```

Comment:
NIT: I saw this kind of code multiple times. Abstract it as a function.

Reply:
Addressed.